src/App.jsx

```diff
@@ -1,10 +1,100 @@
-function App() {
+import { useEffect, useState, useMemo } from 'react';
+import { Container, Box, Typography, Button } from '@mui/material';
+import { SearchBar } from './components/SearchBar/SearchBar';
+import { ArticlesList } from './components/ArticlesList/ArticlesList';
+import { ArticleModal } from './components/ArticleModal/ArticleModal';
+import { SortButton } from './components/SortButton/SortButton';
+
+const App = () => {
+  const [articles, setArticles] = useState([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [open, setOpen] = useState(false);
+  const [sortOrder, setSortOrder] = useState('none'); // 'none', 'ascending', 'descending'
+  const [favoriteArticles, setFavoriteArticles] = useState(() => {
+    const saved = localStorage.getItem('favoriteArticles');
+    return saved ? JSON.parse(saved) : [];
+  });
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const [articlesResponse] = await Promise.all([
+          fetch('http://localhost:3010/articles'),
+        ]);
+        const articlesData = await articlesResponse.json();
+        setArticles(articlesData);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    fetchData();
+  }, [articles]);
+
+  useEffect(() => {
+    localStorage.setItem('favoriteArticles', JSON.stringify(favoriteArticles));
+  }, [favoriteArticles]);
```

Comment:
This shouldn't be in a useEffect.

https://docs.google.com/document/d/1OEp_rSrSa7yilCGILaNYfA0VJuy4Y3kCyPP_uX0yh9o/edit?tab=t.0#heading=h.91rrghh7l0ek

![image](https://github.com/user-attachments/assets/8181f02e-8e70-4e16-90fc-2656c9914df1)